src/lib/Lower.hs

```diff
@@ -559,7 +559,7 @@ traverseFor maybeDest dir ixDict lam@(LamExpr (LamBinder ib ty arr eff) body) =
   ixDict' <- substM ixDict
   ty' <- substM ty
   eff' <- substM $ ignoreHoistFailure $ hoist ib eff
-  let eff'' = extendEffRow (S.singleton IOEffect) eff'
+  let eff'' = extendEffRow (S.singleton InitEffect) eff'
```

Comment:
There is one more reference to IOEffect in this file that you will need to replace, in the code that handles Ix dicts. Actually, we might need to add `RunInit`, as a handler counterpart to `RunIO` that I used in there